### PR TITLE
PLAT-752: add new test for VCachedMemoryRequest

### PR DESCRIPTION
**Pull request by MariaVasilenko**
add test
https://insolar.testrail.io/index.php?/cases/view/5681

### CODE REVIEW

File: ledger-core/virtual/integration/memory_request_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
+package integration
```
Comment (dgovorukhin) [inline ledger-core/virtual/integration/memory_request_test.go:1]: MIssed licence

File: ledger-core/virtual/integration/memory_request_test.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+		{name: "Object state created from pending", precondition: pendingPrecondition},
+	}
+	for _, cases := range testCases {
+		t.Run(cases.name, func(t *testing.T) {
+
+			suite := &memoryCachTest{}
+
+			ctx := suite.initServer(t)
+
+			suite.object = reference.NewSelf(suite.server.RandomLocalWithPulse())
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/memory_request_test.go:59]: suite.server.RandomGlobalWithPulse()

File: ledger-core/virtual/integration/memory_request_test.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+		name         string
+		precondition TestStep
+	}{
+		{name: "Object state created from constructor", precondition: constructorPrecondition},
+		{name: "Object state created from method", precondition: methodPrecondition},
+		{name: "Object state created from pending", precondition: pendingPrecondition},
+	}
+	for _, cases := range testCases {
+		t.Run(cases.name, func(t *testing.T) {
+
```
Comment (dgovorukhin) [inline ledger-core/virtual/integration/memory_request_test.go:54]: Empty line

File: ledger-core/virtual/integration/memory_request_test.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	s.runnerMock.AddExecutionClassify(key, contract.MethodIsolation{
+		Interference: pl.CallFlags.GetInterference(),
+		State:        pl.CallFlags.GetState(),
+	}, nil)
+
+	s.server.SendPayload(ctx, pl)
+}
+
+func (s *memoryCachTest) initServer(t *testing.T) context.Context {
+
```
Comment (dgovorukhin) [inline ledger-core/virtual/integration/memory_request_test.go:147]: empty line

File: ledger-core/virtual/integration/utils/vstatereport.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package utils
```
Comment (dgovorukhin) [inline ledger-core/virtual/integration/utils/vstatereport.go:1]: Missed license

File: ledger-core/virtual/integration/memory_request_test.go [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+	"github.com/insolar/assured-ledger/ledger-core/virtual/handlers"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/integration/mock/publisher/checker"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/integration/utils"
+)
+
+type TestStep func(s *memoryCachTest, ctx context.Context, t *testing.T)
+
+const newState = "new state"
+
+type memoryCachTest struct {
```
Comment (bigbes) [inline ledger-core/virtual/integration/memory_request_test.go:30]: `memoryCacheTest`

File: ledger-core/virtual/integration/memory_request_test.go [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+	s.server.SendPayload(ctx, pl)
+}
+
+func constructorPrecondition(s *memoryCachTest, ctx context.Context, t *testing.T) {
+	pl := utils.GenerateVCallRequestConstructor(s.server)
+	pl.Caller = s.class
+	callOutgoing := pl.CallOutgoing
+
+	result := requestresult.New([]byte(newState), s.object)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/memory_request_test.go:130]: constructor must be with SideEffect

File: ledger-core/virtual/integration/memory_request_test.go [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+		{name: "Object state created from constructor", precondition: constructorPrecondition},
+		{name: "Object state created from method", precondition: methodPrecondition},
+		{name: "Object state created from pending", precondition: pendingPrecondition},
+	}
+	for _, cases := range testCases {
+		t.Run(cases.name, func(t *testing.T) {
+
+			suite := &memoryCachTest{}
+
+			ctx := suite.initServer(t)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/memory_request_test.go:57]: `defer s.server.Stop()`

File: ledger-core/virtual/integration/memory_request_test.go [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+				return false // no resend msg
+			})
+
+			suite.typedChecker.VCachedMemoryResponse.Set(func(resp *payload.VCachedMemoryResponse) bool {
+				require.Equal(t, suite.object, resp.Object)
+				require.Equal(t, newState, resp.Memory)
+				return false
+			})
+
+			executeDone := suite.server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
```
Comment (bigbes) [inline ledger-core/virtual/integration/memory_request_test.go:78]: sure it's execute?

File: ledger-core/virtual/integration/memory_request_test.go (hunk 2 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go]: fixed

File: ledger-core/virtual/integration/memory_request_test.go (hunk 1 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go:6]: done

File: ledger-core/virtual/integration/memory_request_test.go (hunk 3 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go]: fixed

File: ledger-core/virtual/integration/memory_request_test.go (hunk 4 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go]: fixed

File: ledger-core/virtual/integration/utils/vstatereport.go (hunk 5 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/utils/vstatereport.go:6]: fixed

File: ledger-core/virtual/integration/memory_request_test.go (hunk 6 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go]: fixed

File: ledger-core/virtual/integration/memory_request_test.go (hunk 7 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go]: fixed

File: ledger-core/virtual/integration/memory_request_test.go (hunk 8 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go:62]: done

File: ledger-core/virtual/integration/memory_request_test.go (hunk 9 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/memory_request_test.go:83]: fixed

**Review by ElSacramento — APPROVED** [review]

**Review by dgovorukhin — APPROVED** [review]

**Pull request merged**